lib/request.js

```diff
@@ -154,6 +154,24 @@ Request.prototype.end = function (s) {
     }
 };
 
+Request.prototype.setTimeout = function(msecs, callback) {
+  if (callback) this.once('timeout', callback);
+
+  var self = this;
+  function emitTimeout() {
+    self.emit('timeout');
+  }
+ 
+  // set and handle the timeout signals on xhr
+  if (this.xhr) {
+    if (this.timeoutCb) {
+      this.xhr.ontimeout = this.timeoutCb;
```

Comment:
I pretty much ported directly from the node implementation, but used the xhr-provided timeout scaffolding. The `this.once('timeout', callback)` just adds a one-time event handler. It shouldn't call back immediately.

I'm gonna be doing a little more testing with your binding/emit code. It's a lot cleaner.